### ISSUE TITLE
407 Proxy Authentication failure error. We are unable to pass http proxy credentials(username and password)

### ISSUE THREAD

**Issue by SrinivasKG1** [Adyen/adyen-java-api-library#811]
**Describe the bug**
407 Proxy authentication failed error.

**How to reproduce**
There should be http proxy server available.

**Desktop (please complete the following information):**
 - Java Version: jdk1.8.0_161
 - Library Version: Adyenlibrary 18.0.0

**Additional context**
After our upgrade from 13.0.0 to 18.0.0 adyen library our connectivity with adyen is getting failed.

It looks like the updated version is not picking up the http proxy configurations setup at the java startup as below.

java -Dhttps.proxyHost=proxy.com -Dhttps.proxyPort=3128 -Dhttps.nonProxyHosts="*.company.com" -Djdk.http.auth.tunneling.disabledSchemes="" -jar application.war

Can you please advise how we can set the proxy configurations.

-----------------------------------

We tried with the proxy configuration as suggested in https://github.com/Adyen/adyen-java-api-library#proxy-configuration

With this configuration proxy configuration is picked up but proxy Authentication is failing with 407 proxy authentication error.

We tried setting the proxy credentials with the below lines of code, but still it didn't work

Authenticator authenticator = new Authenticator() {

public PasswordAuthentication getPasswordAuthentication() {
return (new PasswordAuthentication("proxyusername", "proxypassword".toCharArray()));
}
};
Authenticator.setDefault(authenticator);

For java 8 and above we also need to have the jdk property set as -Djdk.http.auth.tunneling.disabledSchemes="" for the proxy authentication to work, or else it will again throw the 407-authentication error, even if we pass the proper proxy credentials. We suspect this property might not also be working as expected with the new library.
---------------------------------

To summarize,

Can you please help on how to set the below proxy configurations

proxy host
proxy port
proxy username
proxy password

-Djdk.http.auth.tunneling.disabledSchemes=""

**Comment by jillingk** [comment]
Hi @SrinivasKG1,

Thanks for reaching out to us. I was just trying to take a look, but I cannot find where the problem lies thus far. Have you thought about maybe updating to Java 11? Either way I'm going to pass this on to someone with a little bit more knowledge on the matter and come back to you.

Best, Jilling
Adyen

**Comment by SrinivasKG1** [comment]
@jillingk, thanks for your update, can you provide you suggestion on how we are expected to pass the proxy server's username and password

**Comment by SrinivasKG1** [comment]
@jillingk , Any update on this. Our project is having revenue impact because of this issue.
Can you please help us to resolve this

**Comment by michaelpaul** [comment]
Hello @SrinivasKG1, I've a PR (#815) ready for review to fix this issue. Would you be able to test it in your environment already?

**Comment by SrinivasKG1** [comment]
@michaelpaul is this change published to the maven public repository,  so that I can download the dependency through 
<dependency>
            <groupId>com.adyen</groupId>
            <artifactId>adyen-java-api-library</artifactId>
            <version>18.1.0</version>
        </dependency>

**Comment by michaelpaul** [comment]
@SrinivasKG1 It's not on Maven central yet, but you can fetch it from Github using https://jitpack.io/: 

```xml
    <repositories>
        <repository>
            <id>jitpack.io</id>
            <url>https://jitpack.io</url>
        </repository>
    </repositories>

    <dependencies>
        <dependency>
            <groupId>com.github.adyen</groupId>
            <artifactId>adyen-java-api-library</artifactId>
            <version>PW-7215-SNAPSHOT</version>
            <type>jar</type>
        </dependency>
    </dependencies>
```

**Comment by SrinivasKG1** [comment]
@michaelpaul we tested it with the below properties, but we are still getting the 407 proxy Authentication Failed error

 System.setProperty("jdk.http.auth.tunneling.disabledSchemes", "");
       System.setProperty("http.proxyHost", hostname);
       System.setProperty("http.proxyPort", String.valueOf(port));
       System.setProperty("http.proxyUser", username);
       System.setProperty("http.proxyPassword", password);
       System.setProperty("http.nonProxyHosts", "localhost|127.0.0.1");
       System.setProperty("http.proxySet", "true");
       
       
       System.setProperty("jdk.https.auth.tunneling.disabledSchemes", "");
       System.setProperty("https.proxyHost", hostname);
       System.setProperty("https.proxyPort", String.valueOf(port));
       System.setProperty("https.proxyUser", username);
       System.setProperty("https.proxyPassword", password);
       System.setProperty("https.nonProxyHosts", "localhost|127.0.0.1");
       System.setProperty("https.proxySet", "true");

**Comment by SrinivasKG1** [comment]
@michaelpaul thanks a ton.
It is working now.
You can go head and publish.

Can you please advise, when this will be published in Maven central

**Comment by michaelpaul** [comment]
@SrinivasKG1 Great! We will release this in 18.1.1 as soon as possible.
